File: factorial/c/factorial.c
#include <stdio.h>

/* 
 * Build and run with:
 * `make factorial && ./factorial`
 * or
 * `gcc -o factorial factorial.c && ./factorial`
 */

unsigned int factorial(unsigned int number) {
	unsigned int result = 1;
	for (unsigned int i = 1; i <= number; i++)
		result *= i;
	return result;
}

int main() {
	printf("[");
	for (unsigned int i = 1; i < 10; i++)
		printf("%d,", factorial(i));
	printf("%d]\n", factorial(10));
}

File: factorial/java/Factorial.java
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

class Factorial {
	private static BigInteger factorial(int number) {
		BigInteger result = new BigInteger("1");

		for (int i = 1; i <= number; i++)
			result = result.multiply(new BigInteger(Integer.toString(i)));

		return result;
	}

	public static void main(String[] args) {
		List<BigInteger> factorials = new ArrayList<BigInteger>();

		for (int i = 1; i <= 10; i++)
			factorials.add(factorial(i));

		System.out.println(factorials);
	}
}

